compiler/rustc_mir/src/interpret/operator.rs

```diff
@@ -77,9 +77,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         let res = match bin_op {
             Eq => l == r,
             Ne => l != r,
-            Lt => l < r,
+            Lt => !l & r, // equals l < r
             Le => l <= r,
-            Gt => l > r,
+            Gt => l & !r, // equals l > r
```

Comment:
Will this affect performance or is it any faster?

Reply:
The assembly is the same, (one of the functions even gets removed in the example because its identical to the other one)
https://godbolt.org/z/czfczr

Reply:
If it's the same and there are not difference why not use the original one which is easier to understand? Is there any benefits in doing this?

Reply:
Hm. To me `true & !false` is easier to understand than `true > false`.

Reply:
They are boolean, one has to lookup to know if `false < true`.

Reply:
Ah, no wonder.